### PR TITLE
Update ostorlab CLI: when running scan with cloud runtime and agent group, the procces for creating asset with the api not working

### PR DESCRIPTION
**Pull request by najibraihan**

### CODE REVIEW

**Review by rabsondev — APPROVED** [review]

**Review by mohsinenar — CHANGES_REQUESTED** [review]

File: src/ostorlab/apis/agent_group.py [hunk 1]
```diff
@@ -12,7 +11,10 @@ class CreateAgentGroupAPIRequest(request.APIRequest):
 
     def __init__(self, name: str, description: str, agents: List[AgentType]) -> None:
         """Initializer"""
-        self._name = name
+        if name is None:
```
[the remark this answers is not in the document]
Comment (najibraihan) [reply src/ostorlab/apis/agent_group.py:14]: Change SQLAlchemy version to 1.4, the new version doesn't support whens anymore.

File: setup.cfg [hunk 2]
```diff
@@ -55,10 +55,11 @@ install_requires =
     rich
     ruamel.yaml
     tenacity
-    sqlalchemy
+    sqlalchemy==1.4
```
[the remark this answers is not in the document]
Comment (najibraihan) [reply setup.cfg:58]: Change SQLAlchemy version to 1.4, the new version doesn't support whens anymore.

[the remark this answers is not in the document]
Comment (3asm) [reply setup.cfg:58]: Use ~= to keep receiving fixes.

File: src/ostorlab/apis/assets.py [hunk 3]
```diff
@@ -138,6 +126,23 @@ def __get_asset_variables(self) -> Dict[str, Dict]:
             asset_type_variables = {"file": {"content": None, "path": self._asset.path}}
         elif isinstance(self._asset, domain_name.DomainName):
             asset_type_variables = {"domain": {"domain": self._asset.name}}
+        elif isinstance(self._asset, link.Link):
+            asset_type_variables = {"url": {"urls": [self._asset.url]}}
+        elif isinstance(self._asset, ios_store.IOSStore):
+            # TODO rename package_name to bundle_id in the API
```
Comment (3asm) [inline src/ostorlab/apis/assets.py:132]: Add a ticket or a person for the todo. TODO(xxx):

File: src/ostorlab/apis/runners/authenticated_runner.py [hunk 4]
```diff
@@ -174,3 +176,61 @@ def _sent_request(
...
+            )
+        data: Dict[str, Any] = json.loads(response.content.decode())
+        errors = data.get("errors")
+        if errors is not None and isinstance(errors, list):
+            error = errors[0].get("message")
+            raise runner.ResponseError(f"Response errors: {error}")
+        else:
+            return data
+
+    def _sent_ubjson_request(
```
Comment (3asm) [inline src/ostorlab/apis/runners/authenticated_runner.py:219]: send not sent.

File: src/ostorlab/runtimes/cloud/runtime.py [hunk 5]
```diff
@@ -483,6 +484,16 @@ def _check_agents_exist(
                 return False
         return True
 
+    def _to_serialized(self, value) -> Union[bytes, memoryview, List[Any], str]:
```
Comment (3asm) [inline src/ostorlab/runtimes/cloud/runtime.py:487]: Where is this method used?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/runtimes/cloud/runtime.py (hunk 5 above)
Comment (najibraihan) [reply src/ostorlab/runtimes/cloud/runtime.py:487]: Apply in values of args to solve problem of Byte scalar bug :  "value": self._to_serialized(arg.value),

File: .github/workflows/python-package-lint.yml [hunk 6]
```diff
@@ -38,7 +38,7 @@ jobs:
           options: --check --exclude 'src[\\w\/].+_pb2.py'
       - name: Running linter.
         run: |
-          pylint --rcfile=.pylintrc src/ 
+          pylint --rcfile=.pylintrc -d R1735,W0133 src/
```
Comment (3asm) [inline .github/workflows/python-package-lint.yml:41]: Wtf man!

**Review by 3asm — CHANGES_REQUESTED** [review]

File: .github/workflows/python-package-lint.yml (hunk 6 above)
Comment (najibraihan) [reply .github/workflows/python-package-lint.yml:41]: Fixed

**Review by 3asm — APPROVED** [review]

**Pull request merged**